File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        vector<int> ans;
        
        for (int i = 0; i < nums.size(); i++) {
            int required = target - nums[i];       
            if (mp.find(required) != mp.end()) {
                return {mp[required],i};
            }
            mp[nums[i]] = i;
        }
        return {-1, -1};
    }
};

File: 0001-two-sum/0001-two-sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # Create a list of tuples (number, index)
        nums_with_indices = [(num, i) for i, num in enumerate(nums)]
        
        # Sort the list based on the numbers
        nums_with_indices.sort(key=lambda x: x[0])
        
        left = 0
        right = len(nums) - 1
        
        while left < right:
            current_sum = nums_with_indices[left][0] + nums_with_indices[right][0]
            
            if current_sum == target:
                return [nums_with_indices[left][1], nums_with_indices[right][1]]
            elif current_sum < target:
                left += 1
            else:
                right -= 1
        
        # If no solution is found
        return [-1, -1]